db/1.8/modules/ROOT/pages/database-connector-udt-stored-procedure.adoc

```diff
@@ -0,0 +1,328 @@
+
+[start=4]
+. Click *OK*.
+. In the *Global Elements* view, click *Create* to open the *Choose Global Type* view.
+. In the *Filter* field write `database`, select *Database Config* and click *OK*.
+. In the *Database Config* window, set the *Name* field to `dbConfig`.
+. In the *Connection* field, select *Oracle Connection*.
+. Click *Configure* to add the required Oracle JDBC driver and select either: +
++
+* *Use local file* +
+Install the library using a local file.
+* *Add Maven dependency* +
+Install a Maven dependency to add to the project.
++
+[start=10]
+. In the *Connection* section, set the following parameters: +
++
+* *Host* +
+`localhost`
+* *Port* +
+`1521`
+* *User* +
+`SYS as SYSDBA`
+* *Password* +
+`Oradoc_db1`
+* *Instance* +
+`ORCLCDB`
++
+[start=11]
+. On the *Advanced* tab, set the *Column types* field to *Edit inline*.
+. Click the plus sign (*+*) to add new column types, set the *Id* and *Type name* fields with the following values:
++
+[%header,cols="30a,70a"]
+|===
+| Id | Type name
+| 2003 |  `PEOPLE`
+| 2003 |  `PHONE_NUMBER`
+| 2008 |  `PERSON`
+| 2003 |  `PHONE_ARRAY`
+| 2003 |  `PHONE_BOOK`
+|===
++
+[start=13]
+. Click *Finish* to close the *Column type* window.
+. Click *Test Connection* to confirm that Mule can connect to the database.
+. Click *OK*.
+
+The following screenshot shows the HTTP Listener global element configuration in Studio:
+
+.HTTP Listener configuration
+image::database-oracleudt-example-1.png[HTTP Listener configuration]
+
+The following screenshots shows the Database global element configuration in Studio:
+
+.Database Config General cofiguration
+image::database-oracleudt-example-2.png[Database Config General configuration setting Host, Port, User, Password and Instance values parameters]
+
+.Database Config Advanced cofiguration
+image::database-oracleudt-example-3.png[Database Config advanced configuration setting Column types values]
+
+
+== Create and Run for your Mule App
+
+Create the Mule app flow in Anypoint Studio:
+
+. In the *Mule Palette* view, select the HTTP *Listener* source and drag it on to the canvas. +
+The source listens for incoming HTTP messages attributes.
+. In the *Connector configuration* field, select `HTTP_Listener_config` global configuration.
+. Set the *Path* field to `/oracle`.
+. Drag a *Set Payload* component to the right of *Listener*. +
+This component set a list of items to send to the stored procedure to process.
+. In the *Value* field, set the following code for the list of items: +
++
+[source,xml,linenums]
+----
+[{'name':'Anthony J', 'surname':'Crowley', 'age': 6000, 'phoneNumber': {'areaCode':'020', 'phoneNumber': '777'}},
+	{'name':'A.Z', 'surname':'Fell', 'age': 6000, 'phoneNumber':{'areaCode':'020', 'phoneNumber': '888'}},
+	{'name':'Adam', 'surname':'Young', 'age': 11, 'phoneNumber':{'areaCode':'046', 'phoneNumber': '666'}},
+	{'name':'Anathema', 'surname':'Device', 'age': 27, 'phoneNumber':{'areaCode':'020', 'phoneNumber': '123'}},
+	]
```

Comment:
Are these names supposed to be humorous? 
If so, rethink that: humor does not translate well across cultures.